packages/@aws-cdk/aws-events/lib/connection.ts

```diff
@@ -0,0 +1,449 @@
+   * @default - True
+   */
+  readonly isValueSecret?: string;
+
+  /**
+   * The key for the parameter.
+   */
+  readonly key: string;
+
+  /**
+   * The value associated with the key for the parameter.
+   */
+  readonly value: string;
+}
+
+/**
+ * These are custom parameter to be used when the target is an API Gateway REST APIs or EventBridge ApiDestinations.
+ */
+export interface HttpParameters {
+  /**
+   * Not currently supported by AWS CloudFormation.
+   */
+  // readonly bodyParameters?: Parameter[];
+
+  /**
+   * The headers that need to be sent as part of request invoking the API Gateway REST API or EventBridge ApiDestination.
+   * @default - none
+   */
+  readonly headerParameters?: Parameter[];
+
+  /**
+   * The query string keys/values that need to be sent as part of request invoking the API Gateway REST API or EventBridge ApiDestination.
+   * @default - none
+   */
+  readonly queryStringParameters?: Parameter[];
+}
+
+/**
+ * Contains the OAuth authorization parameters to use for the connection.
+ */
+export interface OAuthParameters {
+  /**
+   * The URL to the authorization endpoint when OAuth is specified as the authorization type.
+   */
+  readonly authorizationEndpoint: string;
+
+  /**
+   * Contains the client parameters for OAuth authorization.
+   */
+  readonly clientParameters: ClientParameters;
+
+  /**
+   * The method to use for the authorization request.
+   */
+  readonly httpMethod: string;
+
+  /**
+   * Contains the OAuth authorization parameters to use for the connection.
+   * @default - none
+   */
+  readonly oAuthHttpParameters?: HttpParameters;
+}
+
+/**
+ * Contains the authorization parameters to use to authorize with the endpoint.
+ */
+export interface AuthParameters {
+  /**
+   * Contains the API key authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly apiKeyAuthParameters?: ApiKeyAuthParameters;
+
+  /**
+   * Contains the Basic authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly basicAuthParameters?: BasicAuthParameters;
+
+  /**
+   * Contains the API key authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly invocationHttpParameters?: HttpParameters;
+
+  /**
+   * Contains the OAuth authorization parameters to use for the connection.
+   * @default - None
+   */
+  readonly oAuthParameters?: OAuthParameters;
+}
+
+/**
+ * The event connection base properties
+ */
+export interface BaseConnectionProps {
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly authorizationType: AuthorizationType;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly authParameters: AuthParameters;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
+   */
+  readonly description: string;
+
+  /**
+   * The name of the connection.
+   *
+   * @default - none
```

Comment:
Why is this `@default - none` ? Either a name is required, in which case there is no `@default` at all, or the name should be generated in which case it's not `none`.

If the name is required, can we make it non-required?

Reply:
Resolved